test-core/src/main/java/io/micronaut/test/support/sql/resolver/CompositeDataSourceResolver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.test.support.sql.resolver;
+
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.test.support.sql.processor.SqlScriptProcessor;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A {@link DataSourceResolver} that delegates to a list of other resolvers.
+ * Available resolvers will be checked in turn, and the first that responds with a datasource will be used.
+ *
+ * @since 4.1.0
+ * @author Tim Yates
+ */
+@Primary
+@Singleton
+@Experimental
+public class CompositeDataSourceResolver implements DataSourceResolver {
```

Comment:
I think we should annotate this with `@Internal` until we know if we want to move this API to the SQL module.